answerking_app/tests/BaseTestClass.py

```diff
@@ -0,0 +1,184 @@
+            stock=100,
+            calories=100,
+        )
+        self.test_item_3: Item = Item.objects.create(
+            name="Chips",
+            price=1.50,
+            description="desc",
+            stock=100,
+            calories=100,
+        )
+
+        self.test_cat_1: Category = Category.objects.create(name="Burgers")
+        self.test_cat_2: Category = Category.objects.create(name="Sides")
+
+        self.test_cat_1.items.add(self.test_item_1)
+        self.test_cat_1.items.add(self.test_item_2)
+        self.status_pending: Status = Status.objects.create(status="Pending")
+        self.status_complete: Status = Status.objects.create(
+            status="Completed"
+        )
+
+        self.test_order_empty: Order = Order.objects.create(
+            address="123A3 Street, Leeds, LS17PP",
+            status=self.status_pending,
+        )
+
+        self.test_order_1: Order = Order.objects.create(
+            address="123 Street, Leeds, LS73PP",
+            status=self.status_pending,
+        )
+
+        self.test_order_2: Order = Order.objects.create(
+            address="456 Test Lane, Bradford, BD30PA",
+            status=self.status_pending,
+        )
+
+        self.test_order_1.order_items.add(
+            self.test_item_1,
+            through_defaults={
+                "quantity": 2,
+                "sub_total": f"{self.test_item_1.price * 2:.2f}",
+            },
+        )
+
+        self.test_order_1.order_items.add(
+            self.test_item_2,
+            through_defaults={
+                "quantity": 1,
+                "sub_total": f"{self.test_item_2.price:.2f}",
+            },
+        )
+        self.test_order_1.calculate_total()
+
+    def tearDown(self):
+        Item.objects.all().delete()
+        Category.objects.all().delete()
+        Status.objects.all().delete()
+
+    def assertJSONResponse(self, expected, actual, response, status_code):
+        self.assertEqual(expected, actual)  # type: ignore[reportGeneralTypeIssues]
+        self.assertEqual(response.status_code, status_code)  # type: ignore[reportGeneralTypeIssues]
+
+    def assertJSONErrorResponse(self, expected, actual, response, status_code):
+        self.assertIsInstance(actual.pop("traceId"), str)  # type: ignore[reportGeneralTypeIssues]
+        self.assertJSONResponse(expected, actual, response, status_code)
+
+    def get_mock_item_api(self, item: Item) -> ItemType:
+        return {
+            "id": item.id,
+            "name": item.name,
+            "price": f"{item.price:.2f}",
+            "description": item.description,
+            "stock": item.stock,
+            "calories": item.calories,
+            "retired": False,
+        }
+
+    def get_mock_category_api(
+        self, category: Category, items: list[ItemType] = []
+    ) -> CategoryType:
+        return {
+            "id": category.id,
+            "name": category.name,
+            "items": items,
+            "retired": False,
```

Comment:
probably would return retired field rather than assuming it is False

Reply:
oh, yeah 👍